crypto/bio/bss_mem.c

```diff
@@ -140,10 +140,16 @@ static int secmem_new(BIO *bi)
 
 static int mem_free(BIO *a)
 {
-    return mem_buf_free(a, 1);
+    BIO_BUF_MEM *bb = (BIO_BUF_MEM *)a->ptr;
```

Comment:
This needs a check for `a == NULL` before dereferencing `a`.
Otherwise optimising compilers will remove the NULL check and cause havoc.